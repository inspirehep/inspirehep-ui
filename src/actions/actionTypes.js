export const SEARCH_REQUEST = 'SEARCH_REQUEST';
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_ERROR = 'SEARCH_ERROR';

export const LITERATURE_REQUEST = 'LITERATURE_REQUEST';
export const LITERATURE_SUCCESS = 'LITERATURE_SUCCESS';
export const LITERATURE_ERROR = 'LITERATURE_ERROR';
export const LITERATURE_REFERENCES_REQUEST = 'LITERATURE_REFERENCES_REQUEST';
export const LITERATURE_REFERENCES_SUCCESS = 'LITERATURE_REFERENCES_SUCCESS';
export const LITERATURE_REFERENCES_ERROR = 'LITERATURE_REFERENCES_ERROR';
export const LITERATURE_AUTHORS_REQUEST = 'LITERATURE_AUTHORS_REQUEST';
export const LITERATURE_AUTHORS_SUCCESS = 'LITERATURE_AUTHORS_SUCCESS';
export const LITERATURE_AUTHORS_ERROR = 'LITERATURE_AUTHORS_ERROR';

export const EXCEPTIONS_REQUEST = 'EXCEPTIONS_REQUEST';
export const EXCEPTIONS_SUCCESS = 'EXCEPTIONS_SUCCESS';
export const EXCEPTIONS_ERROR = 'EXCEPTIONS_ERROR';

export const INSPECT_REQUEST = 'INSPECT_REQUEST';
export const INSPECT_SUCCESS = 'INSPECT_SUCCESS';
export const INSPECT_ERROR = 'INSPECT_ERROR';
