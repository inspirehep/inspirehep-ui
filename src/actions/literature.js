import {
  LITERATURE_ERROR,
  LITERATURE_REQUEST,
  LITERATURE_SUCCESS,
  LITERATURE_REFERENCES_ERROR,
  LITERATURE_REFERENCES_REQUEST,
  LITERATURE_REFERENCES_SUCCESS,
  LITERATURE_AUTHORS_ERROR,
  LITERATURE_AUTHORS_REQUEST,
  LITERATURE_AUTHORS_SUCCESS,
} from './actionTypes';

function fetchingLiterature(recordId) {
  return {
    type: LITERATURE_REQUEST,
    payload: { recordId },
  };
}

function fetchLiteratureSuccess(result) {
  return {
    type: LITERATURE_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureError(error) {
  return {
    type: LITERATURE_ERROR,
    payload: error,
  };
}

function fetchingLiteratureReferences() {
  return {
    type: LITERATURE_REFERENCES_REQUEST,
  };
}

function fetchLiteratureReferencesSuccess(result) {
  return {
    type: LITERATURE_REFERENCES_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureReferencesError(error) {
  return {
    type: LITERATURE_REFERENCES_ERROR,
    payload: error,
  };
}

function fetchingLiteratureAuthors() {
  return {
    type: LITERATURE_AUTHORS_REQUEST,
  };
}

function fetchLiteratureAuthorsSuccess(result) {
  return {
    type: LITERATURE_AUTHORS_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureAuthorsError(error) {
  return {
    type: LITERATURE_AUTHORS_ERROR,
    payload: error,
  };
}

export function fetchLiterature(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiterature(recordId));
    try {
      const response = await http.get(`/literature/${recordId}`, {
        headers: {
          Accept: 'application/vnd+inspire.record.ui+json',
        },
      });
      dispatch(fetchLiteratureSuccess(response.data));
    } catch (error) {
      dispatch(fetchLiteratureError(error.data));
    }
  };
}

export function fetchLiteratureReferences(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiteratureReferences());
    try {
      const response = await http.get(`/literature/${recordId}/references`);
      dispatch(fetchLiteratureReferencesSuccess(response.data));
    } catch (error) {
      dispatch(fetchLiteratureReferencesError(error.data));
    }
  };
}

export function fetchLiteratureAuthors(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiteratureAuthors());
    try {
      const response = await http.get(`/literature/${recordId}/authors`);
      dispatch(fetchLiteratureAuthorsSuccess(response.data));
    } catch (error) {
      dispatch(fetchLiteratureAuthorsError(error.data));
    }
  };
}
