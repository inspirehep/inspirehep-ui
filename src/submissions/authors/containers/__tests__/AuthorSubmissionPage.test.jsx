import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import { getStore, getStoreWithState } from '../../../../fixtures/store';
import AuthorSubmissionPage from '../AuthorSubmissionPage';
import AuthorSubmission from '../../components/AuthorSubmission';
import * as submissions from '../../../../actions/submissions';

jest.mock('../../../../actions/submissions');

describe('AuthorSubmissionPage', () => {
  it('renders with initial store state', () => {
    const store = getStore();
    const wrapper = shallow(<AuthorSubmissionPage store={store} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders with error', () => {
    const store = getStoreWithState({
      submissions: fromJS({
        submitError: {
          message: 'Submit Error',
        },
      }),
    });
    const wrapper = shallow(<AuthorSubmissionPage store={store} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders with query', () => {
    const store = getStoreWithState({
      router: { location: { query: { bai: 'P.Carenza.1' } } },
    });
    const wrapper = shallow(<AuthorSubmissionPage store={store} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('calls submitAuthor on AuthorSubmission submit', () => {
    const store = getStore();
    const formData = {
      name: 'Test',
    };
    const mockSubmitAuthor = jest.fn();
    submissions.submitAuthor = mockSubmitAuthor;
    const wrapper = shallow(<AuthorSubmissionPage store={store} />).dive();

    wrapper.find(AuthorSubmission).simulate('submit', formData);
    expect(mockSubmitAuthor).toHaveBeenCalledWith(formData);
  });
});
