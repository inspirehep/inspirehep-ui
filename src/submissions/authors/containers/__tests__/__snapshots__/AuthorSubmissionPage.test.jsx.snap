// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorSubmissionPage renders with error 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col
    className="mt3 mb3"
    span={14}
  >
    <Row
      className="mb3 pa3 bg-white"
      gutter={0}
    >
      <h3>
        Suggest author
      </h3>
      This form allows you to add new author information. All modifications are transferred to
       
      <ExternalLink
        href="//inspirehep.net/hepnames"
      >
        inspirehep.net/hepnames
      </ExternalLink>
       
      upon approval.
    </Row>
    <Row
      gutter={0}
    >
      <Col>
        <Component
          error={
            Immutable.Map {
              "message": "Submit Error",
            }
          }
          initialFormData={
            Object {
              "bai": undefined,
            }
          }
          onSubmit={[Function]}
        />
      </Col>
    </Row>
  </Col>
</Row>
`;

exports[`AuthorSubmissionPage renders with initial store state 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col
    className="mt3 mb3"
    span={14}
  >
    <Row
      className="mb3 pa3 bg-white"
      gutter={0}
    >
      <h3>
        Suggest author
      </h3>
      This form allows you to add new author information. All modifications are transferred to
       
      <ExternalLink
        href="//inspirehep.net/hepnames"
      >
        inspirehep.net/hepnames
      </ExternalLink>
       
      upon approval.
    </Row>
    <Row
      gutter={0}
    >
      <Col>
        <Component
          error={null}
          initialFormData={
            Object {
              "bai": undefined,
            }
          }
          onSubmit={[Function]}
        />
      </Col>
    </Row>
  </Col>
</Row>
`;

exports[`AuthorSubmissionPage renders with query 1`] = `
<Row
  gutter={0}
  justify="center"
  type="flex"
>
  <Col
    className="mt3 mb3"
    span={14}
  >
    <Row
      className="mb3 pa3 bg-white"
      gutter={0}
    >
      <h3>
        Suggest author
      </h3>
      This form allows you to add new author information. All modifications are transferred to
       
      <ExternalLink
        href="//inspirehep.net/hepnames"
      >
        inspirehep.net/hepnames
      </ExternalLink>
       
      upon approval.
    </Row>
    <Row
      gutter={0}
    >
      <Col>
        <Component
          error={null}
          initialFormData={
            Object {
              "bai": "P.Carenza.1",
            }
          }
          onSubmit={[Function]}
        />
      </Col>
    </Row>
  </Col>
</Row>
`;
